web-services/query/src/main/java/datawave/webservice/query/remote/RemoteQueryServiceImpl.java

```diff
@@ -0,0 +1,347 @@
+                }
+            }
+        }
+    }
+    
+    private String getContent(InputStream content) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        InputStreamReader reader = new InputStreamReader(content, "UTF8");
+        char[] buffer = new char[1024];
+        int chars = reader.read(buffer);
+        while (chars >= 0) {
+            builder.append(buffer, 0, chars);
+            chars = reader.read(buffer);
+        }
+        return builder.toString();
+    }
+    
+    public VoidResponse readVoidResponse(HttpEntity entity) throws IOException {
+        if (entity == null) {
+            return null;
+        } else {
+            VoidResponse response = voidResponseReader.readValue(entity.getContent());
+            if (response.getHasResults()) {
+                return response;
+            } else {
+                throw new RuntimeException(response.getMessages().toString());
+            }
+        }
+    }
+    
+    protected String getBearer(Object callerObject) {
+        init();
+        if (callerObject instanceof DatawavePrincipal) {
+            DatawavePrincipal callerPrincipal = (DatawavePrincipal) callerObject;
+            return "Bearer " + jwtTokenHandler.createTokenFromUsers(callerPrincipal.getName(), callerPrincipal.getProxiedUsers());
+        } else {
+            throw new RuntimeException("Cannot use " + callerObject.getClass().getName() + " as a caller object");
+        }
+    }
+    
+    private <T> T executePostMethodWithRuntimeException(String uriSuffix, Consumer<URIBuilder> uriCustomizer, Consumer<HttpPost> requestCustomizer,
+                    IOFunction<T> resultConverter, Supplier<String> errorSupplier) {
+        init();
+        try {
+            return executePostMethod(uriSuffix, uriCustomizer, requestCustomizer, resultConverter, errorSupplier);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Invalid URI: " + e.getMessage(), e);
+        } catch (IOException e) {
+            config.getFailureCounter().inc();
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+    
+    private <T> T executeGetMethodWithRuntimeException(String uriSuffix, Consumer<URIBuilder> uriCustomizer, Consumer<HttpGet> requestCustomizer,
+                    IOFunction<T> resultConverter, Supplier<String> errorSupplier) {
+        init();
+        try {
+            return executeGetMethod(uriSuffix, uriCustomizer, requestCustomizer, resultConverter, errorSupplier);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Invalid URI: " + e.getMessage(), e);
+        } catch (IOException e) {
+            config.getFailureCounter().inc();
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+    
+    @Override
+    protected String serviceHost() {
+        return config.getServiceHost();
+    }
+    
```

Comment:
Are all these delegate methods really needed? Can't we just get an reference to the configuration object and interact with that directly? (or inject a configuration object where necessary?)

Reply:
I can only get rid of the setter methods because the other methods are required by the parent class.  Creating yet another bean to be configured via the QLF.xml seems more tedious.  I feel the setter methods here are not all that overbearing.  The Idea generated them for me!